docs/proposals/emptydir-pv.md

```diff
@@ -0,0 +1,151 @@
+* The directory backing the PV must get deleted when the PV is deleted.
+
+## Non goals
+* A local storage abstraction that survives extended node outages
+* Snapshotting or simply copying the local volume to another node
+
+## A new PV storage source
+
+As described above, the storage source for the PV must be a local volume. While we already allow hostPath as a PersistentVolumeSource, it is insufficient because:
+* user needs to specify the exact path, thus requires knowledge of the filesystem
+* hard to dynamically provision a hostpath
+* hostpath volumes are not cleaned up
+
+The `emptydir` abstraction solves these problems:
+* it is allocated in `/var/lib/kubelet/resource/uid/`, a location that is predictable but unknow till `resource` creation time
+* The node takes care of allocation, a dynamic provisioner just needs to create the PV
+* it is cleaned up when the pod dies (or in this case when the PV is deleted)
+
+We can create a new [PersistetVolumeSource](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L240) for EmptyDir. Users will request storage by creating PersistetVolumeClaims. If the PVC isn't bound to a PV, one will be created dynamically. The user can't control which node the storage for this PV comes from, or where in the node's filesystem it is allocated. Users can also hand create PVs bound to specific nodes (TODO: elaborate on QoS and what happens if there's no space on the node).
+
+## Scheduling constraints for data gravity
+
+The scheduler will schedule the pod paying attention to any referenced PVs. If an empty dir PV attached to an existing node is found, that trumps all scheduler predicates. Otherwise the scheduler makes a scheduling decision taking into account pod cpu/memory requests (and in the future, requested storage), attaches it to a node, and all subsequent scheduling decisions for any pod referencing the same PV evaluate to the same node. The scheduler "Attaches" a pv to a node by writing an annotation (TBD what actually writes this annotation, scheduler or kubelet, scheduler will at least bind the pod to the node). All pods referencing the PV that enter the "Running" state on the give node will be able to write to the volume. If the node enters the `NotReady` state, all pods referencing the PV attached to the node will remain in `Pending` till such time when an admin either removes the annotation, or rewrites it to point to another node. The directory allocated to the emptydir volume will be exported through the status field of the PV, so admins can yank the disk from a NotReady node and rsync data to the same directory on another node before rewriting the annotation.
+
+The annotations used by the scheduler to attach volumes to nodes will be: `volume.alpha.kubernetes.io/sticky-host`. The kubelet will reject any pod at admission time if it finds that the pod references a PV with this annotation set to a value other than its own node name.
```

Comment:
Just sounds like the hostname label applied to the PV - `alpha.node.kubernetes.io/hostname=SCHEDULED_NODENAME`.  That means it works on any PV